src/main/java/org/terasology/fluid/system/FluidAuthoritySystem.java

```diff
@@ -42,13 +44,29 @@
 public class FluidAuthoritySystem extends BaseComponentSystem {
 
     private static final Logger logger = LoggerFactory.getLogger(FluidAuthoritySystem.class);
+
     @In
     private WorldProvider worldProvider;
     @In
     private FluidRegistry fluidRegistry;
     @In
     private InventoryManager inventoryManager;
 
+    private Optional<Block> getLiquidInReach(final Vector3f start, final Vector3f direction, int distance) {
```

Comment:
Please add a bit of Javadoc, as this method has a contract that's not obious at first sight (the direction is normalized).

```suggestion
	/**
	 * Search for a reachable liquid block in the given direction.
	 * 
	 * A liquid block is reachable iff it is withing the specified distance from the starting location and 
	 * there is not penetrable block preventing direct access.
	 * If the returned option is non-empty, the block is guaranteed to be a liquid block.
	 *
	 * @param start		the starting location
	 * @param direction	the normalized direction vector
	 * @param distance	the reachable distance in number of blocks
	 *
	 * @return option of the liquid block found in reach, empty if none was found
	 */
    private Optional<Block> getLiquidInReach(final Vector3f start, final Vector3f direction, int distance) {
```